### ISSUE TITLE
Custom loss function and regularization

### ISSUE THREAD

**Issue by Berenliu** [dmlc/xgboost#7552]
If I define a custom objective by myself, do the regularization parameters like lambda or alpha still work?
